lit_tests/kernel/wave/scheduling.py

```diff
@@ -120,45 +120,46 @@ def test_gemm_pipelined():
         # CHECK-NEXT: %rotating_reg_1
         # CHECK-NEXT: %rotating_reg_2
         # CHECK-NEXT: %rotating_reg_3
-        # CHECK-NEXT: %read_shared_M_0_N_0_K_0
-        # CHECK-NEXT: %read_shared_M_0_N_0_K_1
+        # CHECK-NEXT: %read_4_shared_M_0_N_0_K_0
+        # TODO: Why did this change from K_1 to K_0???
+        # CHECK-NEXT: %read_2_shared_M_0_N_0_K_0
```

Comment:
This change is weird. The second read here went from K_1 to K_0. The place where its used also change. There are a couple places lower down where something similar happens. K_2 and K_3 both become K_1. I don't understand how my change could have caused this though: it doesn't touch how expansion name suffixes are generated. Unless the logic adding the trailing number suffix was somehow getting invoked for these after expansion? Actually, that might explain it.

Reply:
I have diagnosed this, which I think was the same reason that the dimensions were being written as `K_1` instead of `K:1`. My best guess is that the fx node naming convention by default replaces `:` with `:` and that then it further disambiguates by incrementing a trailing `_\d` if node names conflict. I've instead made scheduling preserve the original node names.